pkg/frontend/admin_openshiftcluster_kubernetespods_logs_test.go

```diff
@@ -0,0 +1,212 @@
+			podName:        "config",
+			containerName:  "container-1",
+			mocks:          func(tt *test, k *mock_adminactions.MockKubeActions) {},
+			wantStatusCode: http.StatusForbidden,
+			wantError:      "403: Forbidden: : Access to the provided namespace 'customer-namespace' is forbidden.",
+			//wantResponse:   []byte(`fake logs` + "\n"),
+		},
+		{
+			name:           "empty container name",
+			namespace:      "openshift-namespace",
+			podName:        "config",
+			containerName:  "",
+			mocks:          func(tt *test, k *mock_adminactions.MockKubeActions) {},
+			wantStatusCode: http.StatusBadRequest,
+			wantError:      "400: InvalidParameter: : The provided container name '' is invalid.",
+			//wantResponse:   []byte(`fake logs` + "\n"),
+		},
+	} {
+		t.Run(fmt.Sprintf("%s: %s", method, tt.name), func(t *testing.T) {
+			resourceID := fmt.Sprintf("/subscriptions/%s/resourcegroups/resourceGroup/providers/Microsoft.RedHatOpenShift/openShiftClusters/resourceName", mockSubID)
+			ti := newTestInfra(t).WithOpenShiftClusters().WithSubscriptions()
+			defer ti.done()
+			k := mock_adminactions.NewMockKubeActions(ti.controller)
+
+			tt.mocks(tt, k)
+
+			ti.fixture.AddOpenShiftClusterDocuments(&api.OpenShiftClusterDocument{
+				Key: strings.ToLower(resourceID),
+				OpenShiftCluster: &api.OpenShiftCluster{
+					ID:   resourceID,
+					Name: "resourceName",
+					Type: "Microsoft.RedHatOpenShift/openshiftClusters",
+					Properties: api.OpenShiftClusterProperties{
+						NetworkProfile: api.NetworkProfile{
+							APIServerPrivateEndpointIP: "0.0.0.0",
+						},
+					},
+				},
+			})
+			ti.fixture.AddSubscriptionDocuments(&api.SubscriptionDocument{
+				ID: mockSubID,
+				Subscription: &api.Subscription{
+					State: api.SubscriptionStateRegistered,
+					Properties: &api.SubscriptionProperties{
+						TenantID: mockTenantID,
+					},
+				},
+			})
+
+			err := ti.buildFixtures(nil)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			f, err := NewFrontend(ctx, ti.audit, ti.log, ti.env, ti.asyncOperationsDatabase, ti.openShiftClustersDatabase, ti.subscriptionsDatabase, api.APIs, &noop.Noop{}, nil, func(*logrus.Entry, env.Interface, *api.OpenShiftCluster) (adminactions.KubeActions, error) {
+				return k, nil
+			}, nil, nil)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			go f.Run(ctx, nil, nil)
+
+			resp, b, err := ti.request(method,
+				fmt.Sprintf("https://server/admin%s/kubernetespodlogs?container=%s&namespace=%s&name=%s", resourceID, tt.containerName, tt.namespace, tt.podName),
+				nil, nil)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			err = validateResponse(resp, b, tt.wantStatusCode, tt.wantError, tt.wantResponse)
+			if err != nil {
+				t.Error(err)
+			}
+		})
+	}
+}
+
+func TestValidateAdminKubernetesPodLogs(t *testing.T) {
+	longName := strings.Repeat("x", 256)
+
+	for _, tt := range []struct {
+		test          string
+		containerName string
+		namespace     string
+		name          string
+		wantErr       string
+	}{
```

Comment:
I had another look and I think we are missing a test cases for invalid container name (to cover `!rxKubernetesString.MatchString(containerName)` portion).

Reply:
Tip: I sometimes introduce code duplication to identify test coverage gaps. Here is an example.

Current coverage without expanding `if`s:

![image](https://user-images.githubusercontent.com/509198/160110612-eee88c2f-f71c-4643-b4d2-fec3a020ec52.png)

Looks like everything is covered. But if we expand `if`s we see this:

![image](https://user-images.githubusercontent.com/509198/160110662-13e307a4-6e82-4c01-87ab-7e80df6122e5.png)